### PR TITLE
Improves some LLNL host configs

### PR DESCRIPTION
**Pull request by kennyweiss**
* Improved PGI support in BLT
* Resolved nvcc + gtest pthread issue (in host-config)

### CODE REVIEW

File: host-configs/llnl-ray-blue_os-clang-coral@2018.05.23.cmake [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+set (CUDA_SEPARABLE_COMPILATION ON CACHE BOOL "" )
+set (CUDA_LINK_WITH_NVCC ON CACHE BOOL "")
+# set the link flags manually since nvcc will link (and not have the wrappers knowledge)
+# on ray - can figure out your equivalant flags by doing mpicc -vvvv
+set (SPECTRUM_ROLLING "/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-rolling-release")
+set (CMAKE_CUDA_LINK_FLAGS "-Xlinker -rpath -Xlinker ${SPECTRUM_ROLLING}/lib -Xlinker -rpath -Xlinker ${COMPILER_HOME}/ibm/lib:/usr/tce/packages/gcc/gcc-4.9.3/lib64 -L${SPECTRUM_ROLLING}/lib/ -lmpi_ibm" CACHE STRING "")
+
+# nvcc does not like gtest's 'pthreads' flag
+set(gtest_disable_pthreads ON CACHE BOOL "")
```
Comment (kennyweiss) [inline host-configs/llnl-ray-blue_os-clang-coral@2018.05.23.cmake]: @rrsettgast -- this line might fix the ``pthreads`` issue you that you encountered.

Comment (davidbeckingsale) [reply host-configs/llnl-ray-blue_os-clang-coral@2018.05.23.cmake]: This issue is specific to CMake 3.12 - but this is a nice fix.

File: tests/internal/CMakeLists.txt [hunk 2]
```diff
@@ -115,7 +115,21 @@ if(ENABLE_GTEST)
...
+    # Tests blt_combine_static_libs macro
+    # Note: Does not currently work on Windows or BGQ
+    ####################################################
+    set(_try_combine_static_libs ON)
+
+    if(CMAKE_HOST_WIN32)
+        set(_try_combine_static_libs OFF)
+    endif()
+
+    if ("$ENV{SYS_TYPE}" STREQUAL "bgqos_0")
```
Comment (kennyweiss) [inline tests/internal/CMakeLists.txt]: I had to disable the ``combine_static_libs`` test on ``bgq`` due to the way that static/shared libraries interact there.  I can create a ticket if anyone thinks this is worth investigating.

File: host-configs/llnl-ray-blue_os-clang-coral@2018.05.23.cmake (hunk 1 above)
Comment (kennyweiss) [reply host-configs/llnl-ray-blue_os-clang-coral@2018.05.23.cmake]: I think I also ran into this problem with ``cmake/3.9.2``

Comment (davidbeckingsale) [reply host-configs/llnl-ray-blue_os-clang-coral@2018.05.23.cmake]: 🤔

**Pull request merged**